src/Components/DynamicTable.js

```diff
@@ -0,0 +1,77 @@
+import React, {Component} from 'react';
+import contacts from '../contacts.json';
+import './DynamicTable.css'
+
+class DynamicTable extends Component{
+
+    state = {contactsState: contacts.slice(0,5)}
+
+    addRandom(){
+        let randomContact = contacts[Math.floor(Math.random()*contacts.length)]
+        this.setState({contactsState: [...this.state.contactsState, randomContact]})
+    }
+
+    sortByName(){
```

Comment:
Hey Emad, this works fine but for the sake of making your code more readable I would recommend writing your function in a way where instead of having  the sorting happen at the same time as you update the state breaking this in two separate actions
```
sortByName(){
       let sortedContacts = this.state.contactsState.sort((a, b) => { 
            let nameA = a.name.toUpperCase(); 
            let nameB = b.name.toUpperCase(); 
            if (nameA < nameB) {
              return -1;
            }
            if (nameA > nameB) {
              return 1;
            }
            return 0;
        })

        this.setState({contactsState: sortedContacts})     
    }